### PR TITLE
Add Neural Network distribution output layer and loss functions

### PR DESCRIPTION
**Pull request by oxinabox**
I wrote this in Lux.jl (first time using that) rather than Flux.jl,
so in the future it should be trivial to incorporate this into a NeuralODE etc and fit it against (or fine tune it with) measured values from a ODE model.

This came out fairly elegant i think

### CODE REVIEW

File: src/neural_net.jl [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    sample_weight::Float64 = 1.0
+    mean_std_weight::Float64 = 2.0  # 2x as much information as a single point
+end
+
+
+
+(loss::DistributionLoss)(ŷs, ys) = mean(((ŷ, y),)->loss1(loss, ŷ, y), zip(ŷs, ys))
+
+# for samples
+loss1(d::DistributionLoss, dist::Distribution, y::Number) = d.sample_weight * -loglikelihood(dist, y)
```
Comment (oxinabox) [inline src/neural_net.jl]: This needs to be changed to

```suggestion
loss1(d::DistributionLoss, dist::Distribution, y::Number) = d.sample_weight * abs2(mean(dist) - mean(y))
```
because we often have single samples, and loglikelhood will overfit them down to a single point delta distribution
